### ISSUE TITLE
out of memory kill

### ISSUE THREAD

**Issue by 1997cui** [ledgerwatch/erigon#8485]
#### System information

Erigon version: `./erigon --version`

OS & Version: Linux

Commit hash: 2.52.5-7c5e4e77

Erigon Command (with flags/config):
```
Environment="GOGC=50"
Environment="GOMEMLIMIT=6000MiB"
Environment="GOMAXPROCS=4"
WorkingDirectory=%h/software/blockchain

%h/software/blockchain/erigon --pprof --torrent.download.rate=80mb --datadir=/srv/blockchain/erigon/ --http.api eth,erigon,trace,debug,web3,net,txpool,ots,engine --http.corsdomain "*" --batchSize=8mb --ws --internalcl
```
Concensus Layer: Internal

Concensus Layer Command (with flags/config): Internal

Chain/Network: 

#### Expected behaviour

Consistent memory usage

#### Actual behaviour

It seems that memory is leaking.

#### Steps to reproduce the behaviour
![mem2](https://github.com/ledgerwatch/erigon/assets/5462776/b37b35ab-9b39-4ba7-b04c-d31cad2fc962)

#### Backtrace

````
[backtrace]
````

**Comment by AskAlexSharov** [comment]
thank you for good bug-report: here is 1 part of GOMEMLIMIT support https://github.com/ledgerwatch/erigon/pull/8498

still need investigate why sendBlocksRequest eating much

**Comment by luarx** [comment]
Just in case it can help, **we are seeing also a huge memory consumption since we upgraded** on October 10th two of our Erigon nodes from `v2.42.0 -> v2.50.1` 😨 

*More info: We are using Lighthouse for Consesus layer
![image](https://github.com/ledgerwatch/erigon/assets/22997139/b9c444ad-74a7-4eb5-a14b-271110f46847)
![image](https://github.com/ledgerwatch/erigon/assets/22997139/886c01df-8ab3-46ae-8fab-ed01163b47bd)

**Comment by AskAlexSharov** [comment]
1. can try cherry-pick https://github.com/ledgerwatch/erigon/pull/8431
2. add flag --pprof
go tool pprof -inuse_space -png http://127.0.0.1:6060/debug/pprof/heap > mem.png
show this file

**Comment by luarx** [comment]
Sorry, I use Docker images :(

**Comment by 1997cui** [comment]
Hi I cherry-picked #8431 and #8498 , run for 3 days, and here is the latest result:
![mem](https://github.com/ledgerwatch/erigon/assets/5462776/ca906756-2c5c-4b13-837c-f689e13dcb6c)
Here is my git history
```bash
ctyi@cuiDesktop:~/software/blockchain/erigonrepo$ git log --oneline 
9a652126c (HEAD -> 2.52-try-fix-mem) save
ca73eae9d bor: demote some logs
fe1fa86fb sync: refactor headerdownload.Link pruning operations
100f04007 sync: fix a memory leak when header verification fails
8e94473e7 sync: log invalid and rejected header stats
7c5e4e770 (tag: v2.52.5, origin/release/2.52, release/2.52) Patch 2.52.5 (#8468)
48a109950 (tag: v2.52.4) Set root-reserve-mb to 16GB
fe5fdda1e (tag: v2.52.3) Set root-reserve-mb to 4GB
d8ed8e68e (tag: v2.52.2) goreleaser: Maximize build space (#8451)
6f793b5c7 (tag: v2.52.1) Fix invalid pre-fetched header broadcast (#8442) (#8446)
24ee988ea Patch release 2.52.1 (#8440)
761af94b2 (tag: v2.52.0) params: remove dev from v2.52.0 (#8432)

```
